packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -299,13 +300,31 @@ const Dropdown = React.forwardRef(
         : {} // When autoAlign is turned off, floating-ui will not be used
     );
 
+    const { reference, floating } = elements;
+
+    const updatePosition = () => {
+      computePosition(reference as ReferenceElement, floating as HTMLElement, {
+        middleware: [hide({ strategy: 'referenceHidden' })],
+      }).then(({ middlewareData }) => {
+        if (middlewareData.hide) {
+          floating &&
+            Object.assign(floating.style, {
+              visibility: middlewareData.hide.referenceHidden
+                ? 'hidden'
+                : 'visible',
+            });
+        }
+      });
+    };
+
     useEffect(() => {
       if (autoAlign) {
         Object.keys(floatingStyles).forEach((style) => {
```

Comment:
If we get the middleware working with the hook, the visibility style could be applied by adding it to `floatingStyles` here before we iterate over it.

```suggestion
      if (autoAlign) {
        Object.keys({...floatingStyles, visibility: middlewareData.hide?.referenceHidden ? 'hidden' : 'visible'}).forEach((style) => {
```

Or maybe on the preceding line so it's less dense